dhcpv4/dhcpv4.go

```diff
@@ -500,6 +568,18 @@ func (d *DHCPv4) Summary() string {
 	)
 	ret += "  options=\n"
 	for _, opt := range d.options {
+		// Parse and display sub-options
+		if opt.Code == OptionVendorSpecificInformation {
```

Comment:
I understand that because of this block of code, moving the BSDP specific code above may be difficult. Instead of giving up though, I'd rather modify the signature of `RequestFromOffer` to be variadic and accept optional functions that will modify the DHCP options. Something like this, https://play.golang.org/p/DrYLFweJNjJ but for DHCP options

Reply:
I'm not sure I follow - do you mean modify the `Summary()` function instead?

Reply:
Sorry, I was tricked by the compressed view on GitHub :( let me re-read this

Reply:
Yeah, I thought I was reading `RequestFromOffer`, not `Summary`.

By the way, same principle - `BSDPString()` is too vendor-specific to live in the `dhcpv4 `package. In the `dhcpv6` package I made `Options` an interface, not a struct, which makes it a lot more flexible - you just call `myopt.String()` and it will print your option-specific code. Unfortunately the `dhcpv4` package is a bit behind.

In this case I'd rather remove the BSDPString method and the loop below that prints the suboption, and wait until, after your PR, we have `Option` as interface. I commit to do this as soon as this PR is merged so we can quickly re-add this feature.

To summarize: remove the BSDPString method and the loop that prints the suboption in `Summary`, and please add a TODO for me to implement `Option` as interface